test/gtest/ucp/test_ucp_ep_reconfig.cc

```diff
@@ -173,36 +207,57 @@ void test_ucp_ep_reconfig::entity::store_config()
     }
 
     m_cfg_index = ep()->cfg_index;
+
+    /* Calculate number of reused lanes */
+    auto num_reused    = (ucp_ep_num_lanes(ep()) / 2) / num_paths();
+    auto test          = static_cast<const test_ucp_ep_reconfig*>(m_test);
+    m_num_reused_lanes = test->reuse_lanes() ? num_reused : 0;
 }
 
-ucp_tl_bitmap_t test_ucp_ep_reconfig::entity::ep_tl_bitmap() const
+ucp_tl_bitmap_t
+test_ucp_ep_reconfig::entity::ep_tl_bitmap(unsigned max_num_devs) const
 {
-    if (ep() == NULL) {
-        return UCS_STATIC_BITMAP_ZERO_INITIALIZER;
-    }
-
     ucp_tl_bitmap_t tl_bitmap = UCS_STATIC_BITMAP_ZERO_INITIALIZER;
-    for (ucp_lane_index_t lane = 0; lane < ucp_ep_num_lanes(ep()); ++lane) {
+    unsigned dev_count        = 0;
+
+    for (auto lane = 0; lane < ucp_ep_num_lanes(ep()); ++lane) {
+        if (ucp_ep_get_path_index(ep(), lane) > 0) {
+            continue;
+        }
+
+        if (dev_count++ >= max_num_devs) {
+            break;
+        }
+
         UCS_STATIC_BITMAP_SET(&tl_bitmap, ucp_ep_get_rsc_index(ep(), lane));
     }
 
     return tl_bitmap;
 }
 
+ucp_tl_bitmap_t test_ucp_ep_reconfig::entity::reduced_tl_bitmap() const
+{
+    if ((ep() == NULL) || !m_exclude_ifaces) {

```

Comment:
why would ep be null here?

Reply:
because it's being initialized in `ucp_ep_create_to_worker_addr` which is not yet called in this function